examples/text_labels_and_annotations/text_rotation.py

```diff
@@ -0,0 +1,51 @@
+"""
+===================================
+Default text rotation demonstration
+===================================
+
+The way Matplotlib does text layout by default is counter-intuitive to some, so
+this example is designed to make it a little clearer.
+
+The text is aligned by its bounding box (the rectangular box that surrounds the
+ink rectangle).  The order of operations is rotation then alignment.
+Basically, the text is centered at your x,y location, rotated around this
+point, and then aligned according to the bounding box of the rotated text.
+
+So if you specify left, bottom alignment, the bottom left of the
+bounding box of the rotated text will be at the x,y coordinate of the
+text.
+
+But a picture is worth a thousand words!
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def addtext(ax, props):
+    ax.text(0.5, 0.5, 'text 0', props, rotation=0)
```

Comment:
How about putting this in the loop too? `for i, angle in enumerate([0, 45, 135, 225, -45]):`